pkg/docker/docker.go

```diff
@@ -318,7 +331,7 @@ func getDefaultContext(timeout time.Duration) (context.Context, context.CancelFu
 // GetImageWorkdir returns the WORKDIR property for the given image name.
 // When the WORKDIR is not set or empty, return "/" instead.
 func (d *stiDocker) GetImageWorkdir(name string) (string, error) {
-	resp, err := d.kubeDockerClient.InspectImage(name)
+	resp, err := d.InspectImage(name)
```

Comment:
the kubeclient inspectimage behavior changed significantly (which is why this pull includes its own implementation of inspectimage now).  as an aside, in talking to @smarterclayton about it, he thinks we probably should not have migrated to the kubeclient interfaces, unfortunately.  So even assuming I can get this to a working state as is, i think in the near future we're going to want to revisit the s2i changes to either:

1) go back to fsouza/go-dockerclient
2) move off kubeclient and only depend on the docker-engine apis

because we're going to be continuously vulnerable to kubeclient changing out from underneath us and being unable to bump the s2i code into origin as a result (which is what happened here).

it's fairly urgent since @jim-minter and @PI-Victor both have pending changes to s2i that we need to get into origin, and that can't happen until we get s2i back to a state where it can be bumped in origin.